include/GooseEYE/GooseEYE.h

```diff
@@ -36,11 +36,17 @@ enum class path_mode
 
 // -------------------------------------------------------------------------------------------------
 
-xt::xarray<int> path(
-  const std::vector<int> &xa,
-  const std::vector<int> &xb,
+xt::xtensor<size_t,2> path(
+  const xt::xtensor_fixed<size_t, xt::xshape<3>>& x0,
+  const xt::xtensor_fixed<size_t, xt::xshape<3>>& x1,
```

Comment:
Consider using:

```cpp
template <class T>
xt::xtensor<size_t,2> path(
  const T&& x0, 
  const T&& x1
```
The disadvantage if this though is the Python interface: it would need some wrapper code.